### PR TITLE
Improve documentation for the SpecialResource API

### PR DESCRIPTION
**Pull request by qbarrand**
Mark unused types as such.
Re-generate manifests.

/cc @pacevedom @pmtk

### CODE REVIEW

File: api/v1beta1/specialresource_types.go [hunk 1]
```diff
@@ -54,94 +58,127 @@ type SpecialResourceArtifacts struct {
...
 	// +kubebuilder:validation:Optional
 	Debug bool `json:"debug"`
+
+	// Set are Helm hierarchical values for this chart installation.
 	// +kubebuilder:validation:Optional
 	// +kubebuilder:pruning:PreserveUnknownFields
 	// +kubebuilder:validation:EmbeddedResource
 	Set unstructured.Unstructured `json:"set,omitempty"`
+
+	// DriverContainer is not used.
```
Comment (pmtk) [inline api/v1beta1/specialresource_types.go:116]: It's not used in code, but I've seen it used just like a values from `Set` would be:
https://github.com/openshift-psap/special-resource-operator/blob/master/charts/example/simple-kmod-0.0.1/simple-kmod.yaml#L20
https://github.com/openshift-psap/special-resource-operator/blob/master/charts/example/simple-kmod-0.0.1/templates/0000-buildconfig.yaml#L28

TBH this field can give a bad impression that it actually does anything. Breaking changes (for partners) aside, I guess it could be removed and `Set` used instead

Comment (pacevedom) [reply api/v1beta1/specialresource_types.go:116]: If it is not used in the code it is safe to remove even if not defined. Since we are using OpenAPI we take advantage of `additionalProperties` defaulting to null, which is equivalent to `true`, which means you can add whatever you want in an instance for this definition. It is going to be ignored later on.
Adding/removing mandatory files does need a new API version, though.
This depends on what our versioning strategy is in regards to optionals. I would change to v1beta2 and deprecate the old one, but that might be noisy with partners.

Comment (qbarrand) [reply api/v1beta1/specialresource_types.go:116]: Agreed. I'm looking into migrating the API to a new version with all these fields removed.

File: config/crd/bases/sro.openshift.io_specialresources.yaml [hunk 2]
```diff
@@ -216,14 +269,20 @@ spec:
...
                 type: boolean
               namespace:
+                description: Namespace describes in which namespace the chart will
+                  be installed.
                 type: string
               nodeSelector:
                 additionalProperties:
                   type: string
+                description: NodeSelector is used to determine on which nodes the
+                  soiftware stack should be installed.
```
Comment (ukalifon) [inline config/crd/bases/sro.openshift.io_specialresources.yaml]: typo: should be "software"

Comment (qbarrand) [reply config/crd/bases/sro.openshift.io_specialresources.yaml]: Fixed, thanks!

File: api/v1beta1/specialresource_types.go [hunk 3]
```diff
@@ -54,94 +58,127 @@ type SpecialResourceArtifacts struct {
...
 	Artifacts SpecialResourceArtifacts `json:"artifacts,omitempty"`
 }
 
-// SpecialResourceSpec defines the desired state of SpecialResource
+// SpecialResourceSpec describes the desired state of the resource, such as the recipe to be used and a selector
+// on which nodes it should be installed.
+// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#spec-and-status
+// +kubebuilder:validation:Required
 type SpecialResourceSpec struct {
+	// Chart describes the Helm chart that needs to be installed.
```
Comment (pacevedom) [inline api/v1beta1/specialresource_types.go:94]: Not sure about the duality of chart vs recipe here and there. Maybe we should use the same word?

File: api/v1beta1/specialresource_types.go [hunk 4]
```diff
@@ -54,94 +58,127 @@ type SpecialResourceArtifacts struct {
...
+
+	// Namespace describes in which namespace the chart will be installed.
 	// +kubebuilder:validation:Required
 	Namespace string `json:"namespace"`
+
+	// ForceUpgrade is not used.
 	// +kubebuilder:validation:Optional
 	ForceUpgrade bool `json:"forceUpgrade"`
+
+	// If Debug is true, additional debugging output will be written in the log file.
```
Comment (pacevedom) [inline api/v1beta1/specialresource_types.go]: I would rephrase it to "Debug enables additional logging"

File: api/v1beta1/specialresource_types.go [hunk 5]
```diff
@@ -54,94 +58,127 @@ type SpecialResourceArtifacts struct {
...
 	// +kubebuilder:validation:Optional
 	Debug bool `json:"debug"`
+
+	// Set are Helm hierarchical values for this chart installation.
```
Comment (pacevedom) [inline api/v1beta1/specialresource_types.go]: Maybe "Set is a user defined hierarchical value tree from where the recipe can take parameters"

File: api/v1beta1/specialresource_types.go (hunk 3 above)
Comment (qbarrand) [reply api/v1beta1/specialresource_types.go:94]: [OpenShift docs](https://docs.openshift.com/container-platform/4.9/hardware_enablement/psap-special-resource-operator.html) use `chart`, so let us do that here as well.

File: api/v1beta1/specialresource_types.go (hunk 5 above)
Comment (qbarrand) [reply api/v1beta1/specialresource_types.go]: Done, thanks.

File: api/v1beta1/specialresource_types.go (hunk 4 above)
Comment (qbarrand) [reply api/v1beta1/specialresource_types.go]: Done, thanks.

**Comment by qbarrand** [comment]
/ok-to-test
@pacevedom @pmtk PTAL

**Review by pacevedom — APPROVED** [review]
